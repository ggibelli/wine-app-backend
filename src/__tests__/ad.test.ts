import gql from 'graphql-tag';
import createTestServer from '../tests/helper';
import { createTestClient } from 'apollo-server-testing';
// import Ads from '../data-sources/ads';
import { ads as adsMocks } from '../tests/adsMocks';
// import { AdTest } from '../tests/adsMocks';
//import { TypeProduct, TypeAd } from '../types';
const AD = gql`
  {
    ads(typeAd: SELL, typeProduct: ADWINE) {
      content
    }
  }
`;

describe('queries', () => {
  test('ads', async () => {
    const { server, ads } = createTestServer({
      context: () => ({ user: { id: 1, email: 'a@a.a' } }),
    });
    //@ts-ignore
    ads.getAds = jest.fn(() => adsMocks());
    const { query } = createTestClient(server);

    const res = await query({ query: AD });
    expect(res).toMatchSnapshot();
  });
});
