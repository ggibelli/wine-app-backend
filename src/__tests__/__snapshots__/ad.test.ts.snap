// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queries ads 1`] = `
Object {
  "data": Object {
    "ads": Array [
      Object {
        "content": "wow",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
